packages/dev/serializers/src/glTF/2.0/glTFExporter.ts

```diff
@@ -448,1902 +429,1145 @@ export class _Exporter {
+                }
             }
 
-            const imagePaddingBuffer = new ArrayBuffer(imagePadding);
-            const imagePaddingView = new Uint8Array(imagePaddingBuffer);
-            for (let i = 0; i < imagePadding; ++i) {
-                imagePaddingView[i] = 0;
-            }
+            // Nodes that use this skin.
+            const skinedNodes = this._nodesSkinMap.get(skin);
 
-            const glbData = [headerBuffer, jsonChunkBuffer, binaryChunkBuffer, binaryBuffer];
+            // Only create skeleton if it has at least one joint and is used by a mesh.
+            if (skin.joints.length > 0 && skinedNodes !== undefined) {
+                // create buffer view for inverse bind matrices
+                const byteStride = 64; // 4 x 4 matrix of 32 bit float
+                const byteLength = inverseBindMatrices.length * byteStride;
+                const bufferViewOffset = this._dataWriter.byteOffset;
+                const bufferView = createBufferView(0, bufferViewOffset, byteLength, undefined);
+                this._bufferViews.push(bufferView);
+                const bufferViewIndex = this._bufferViews.length - 1;
+                const bindMatrixAccessor = createAccessor(bufferViewIndex, AccessorType.MAT4, AccessorComponentType.FLOAT, inverseBindMatrices.length, null, null);
+                const inverseBindAccessorIndex = this._accessors.push(bindMatrixAccessor) - 1;
+                skin.inverseBindMatrices = inverseBindAccessorIndex;
+                inverseBindMatrices.forEach((mat) => {
+                    mat.m.forEach((cell: number) => {
+                        this._dataWriter.writeFloat32(cell);
+                    });
+                });
 
-            // binary data
-            for (let i = 0; i < this._orderedImageData.length; ++i) {
-                glbData.push(this._orderedImageData[i].data);
+                this._skins.push(skin);
+                for (const skinedNode of skinedNodes) {
+                    skinedNode.skin = this._skins.length - 1;
+                }
             }
+        }
+    }
 
-            glbData.push(binPaddingBuffer);
+    private async _exportSceneAsync(): Promise<void> {
+        const scene: IScene = { nodes: [] };
 
-            glbData.push(imagePaddingBuffer);
+        // Scene metadata
+        if (this._babylonScene.metadata) {
+            if (this._options.metadataSelector) {
+                scene.extras = this._options.metadataSelector(this._babylonScene.metadata);
+            } else if (this._babylonScene.metadata.gltf) {
+                scene.extras = this._babylonScene.metadata.gltf.extras;
+            }
+        }
 
-            const glbFile = new Blob(glbData, { type: "application/octet-stream" });
+        //  TODO:
+        //  deal with this from the loader:
+        //  babylonMaterial.invertNormalMapX = !this._babylonScene.useRightHandedSystem;
+        //  babylonMaterial.invertNormalMapY = this._babylonScene.useRightHandedSystem;

```

Comment:
TODO

Reply:
This one does need to be dealt with, but after this PR.

In general, should we avoid TODOs like this?

Reply:
As long as we are tracking it, I'm okay with leaving a TODO.